src/main/java/infinityitemeditor/events/PlayerInteractHandler.java

```diff
@@ -0,0 +1,31 @@
+package infinityitemeditor.events;
+
+import infinityitemeditor.data.DataItem;
+import net.minecraft.client.Minecraft;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.Items;
+import net.minecraft.network.play.client.CCreativeInventoryActionPacket;
+import net.minecraftforge.event.entity.player.PlayerInteractEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class PlayerInteractHandler {
+
+    public PlayerInteractHandler() {
+        this.minecraft = Minecraft.getInstance();
+    }
+
+    private final Minecraft minecraft;
+
+    @SubscribeEvent
+    public void onInteract(PlayerInteractEvent.EntityInteractSpecific e) {
+        DataItem heldItemStack = new DataItem(e.getPlayer().getMainHandItem());
+        Item heldItem = e.getPlayer().getMainHandItem().getItem();
+        if (e.getPlayer().isCreative() && e.getTarget().getType() == EntityType.PLAYER && e.getPlayer().isShiftKeyDown() && (heldItem == Items.PLAYER_HEAD) && heldItemStack.getTag().getSkullOwner().get() == null) {
```

Comment:
No need for extra parenthesis when they are all &&

Try to put the checks that you think are most likely to fail at the beginning. For example checking if player is holding shift. That way in most cases, it will only check that and not continue to check the other values. As long as one of them are false, it will all be false. So try to "reach false as soon as possible".

Rather than checking if shift is held, you should probably check if the player is sneaking.